zcash_client_backend/src/data_api/wallet/input_selection.rs

```diff
@@ -239,6 +239,75 @@ impl<FeeRuleT, NoteRef> Proposal<FeeRuleT, NoteRef> {
     pub fn is_shielding(&self) -> bool {
         self.is_shielding
     }
+
+    /// Returns whether or not this proposal requires interaction with the specified pool
+    //
+    // TODO: the `TransactionRequest` type doesn't contain quite enough information about
+    // recipients, because it does not resolve each output payment address to a single receiver.
+    // In the presence of source restriction metadata or pool-crossing prohibitions, we need to
+    // resolve receiving pools prior to constructing the proposal; otherwise we don't have
+    // sufficient information to fully determine the privacy implications of a transaction.
```

Comment:
Since this is resolved by #1187, let's make sure it doesn't remain after both are merged.